vllm/worker/habana_model_runner.py

```diff
@@ -172,11 +172,16 @@ def generate_prompt_buckets(bs_bucket_config,
 def generate_decode_buckets(bs_bucket_config, blocks_bucket_config,
                             max_blocks):
     buckets = []
-    for bs in warmup_range(bs_bucket_config):
-        for blocks in warmup_range(blocks_bucket_config):
+    bs_buckets = warmup_range(bs_bucket_config)
+    block_buckets = warmup_range(blocks_bucket_config)
+    bmin, bstep, bmax = blocks_bucket_config
+    last_bucket = max_blocks if (max_blocks // bstep
+                                 == 0) else (max_blocks // bstep + 1) * bstep
```

Comment:
Is this condition ok?

max_blocks // bstep == 0 means that max_blocks < bstep since it's integer division. In that case we should still warm up to 'step' and not 'max_blocks'

(max_blocks // bstep + 1) * bstep won't work if max_blocks is divisible by bstep. For example: max_blocks = 1024, bstep=128 => (1024 // 128 + 1) * 128 = 1152

Couldn't we simply use:
```python
last_bucket = round_up(max_blocks, bstep)
```
?

Reply:
done